### PR TITLE
universal rules update rake task

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
idempotent CSV uploader for universal rules and concepts

## WHY
So that we can update these records in a standardized and safe manner (and prep the way for a GUI interface later)

## HOW
new rake task 

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Update-the-concepts-feedback-for-universal-rules-in-the-internal-tool-ccc6d55ea35845bab690f1dae966111c

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | yes

### CODE REVIEW

**Comment by happythenewsad** [comment]
@anathomical @emilia-friedberg

File: services/QuillLMS/app/lib/universal_rule_loader.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        puts "\nCannot find universal #{type} Rule with UID #{row['Rule UID']}. Creating new rule."
+        rule = Evidence::Rule.new(
+          uid: row['Rule UID'],
+          name: row['Rule'],
+          universal: true,
+          rule_type: type,
+          suborder: row['Overall Rule Priority'],
+          optimal: true, # Hardcoded, b/c the input file does not currently include an 'optimal' column
+          state: Evidence::Rule::STATES.first # same as above
+        )
```
Comment (anathomical) [inline services/QuillLMS/app/lib/universal_rule_loader.rb:44]: If this fires because of the second two conditions (not universal or not the right type) you'll attempt to create a new rule with the same UID which will fail due to uniqueness validation.

File: services/QuillLMS/spec/lib/universal_rule_loader_spec.rb [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        rule = Evidence::Rule.find_by_uid '1d66a'
+        expect(rule.concept_uid).to eq '4d5e3'
+
+        feedback = Evidence::Feedback.first 
+        expect([feedback.rule_id, feedback.text]).to eq(
+          [rule.id, 'the feedback']
+        )
+      end
+
+      it 'should create a new rule and new feedback' do
```
Comment (anathomical) [inline services/QuillLMS/spec/lib/universal_rule_loader_spec.rb:90]: Need to figure out what behavior we want when there's an existing rule with the UID, but it's not universal or is the wrong type and add some test cases for that edge case behavior (which might just be to throw an error or something).

**Review by anathomical — CHANGES_REQUESTED** [review]
Need to figure out what behavior we actually want if there's a rule with the specified UID but it's somehow not universal or has the wrong type.

**Review by emilia-friedberg — APPROVED** [review]
LGTM other than Thomas's points, don't need to review again when those are made.

**Review by anathomical — APPROVED** [review]
